File: src/windows/magma/app/StringValue.c
#include "./StringValue.h"
class StringValue implements Value {
	mut value: &[I8];
	constructor (mut value: &[I8]) {
		this.value = value;
	}
	mut generate(): &[I8] {
		return "\"" + this.value + "\"";
	}
	mut toValue(): Option<Value> {
		return new Some<Value>(this);
	}
	mut findChild(): Option<Value> {
		return new None<Value>();
	}
	mut resolve(state: CompileState): Type {
		return Primitive.Unknown;
	}
	mut generateAsEnumValue(structureName: &[I8]): Option<&[I8]> {
		return new None<&[I8]>();
	}
}

File: src/node/magma/app/StringValue.ts
import { Value } from "../../magma/app/Value";
import { Option } from "../../magma/api/option/Option";
import { Some } from "../../magma/api/option/Some";
import { None } from "../../magma/api/option/None";
import { Type } from "../../magma/api/Type";
import { CompileState } from "../../magma/app/compile/CompileState";
import { Primitive } from "../../magma/app/Primitive";
class StringValue implements Value {
	value: string;
	constructor (value: string) {
		this.value = value;
	}
	generate(): string {
		return "\"" + this.value + "\"";
	}
	toValue(): Option<Value> {
		return new Some<Value>(this);
	}
	findChild(): Option<Value> {
		return new None<Value>();
	}
	resolve(state: CompileState): Type {
		return Primitive.Unknown;
	}
	generateAsEnumValue(structureName: string): Option<string> {
		return new None<string>();
	}
}
